### ISSUE TITLE
Option to remove duplicates when same error is reported from multiple linters

### ISSUE THREAD

**Issue by ramya-rao-a** [alecthomas/gometalinter#286]
_From @daenney on May 29, 2017 20:42_

When an issue is detected in a file it is reported in the problems tab by a number of linters gometalinter uses. This results in a tremendous amount of issues reported for what is only one thing:

<img width="604" alt="screen shot 2017-05-29 at 22 39 58" src="https://cloud.githubusercontent.com/assets/569574/26561576/caaa9314-44bf-11e7-9a1e-a566c2bea065.png">

The issue should be reproducible with a snippet like this:

```go
package main

import "flag"

func main() {
    flag.Bool()
}
```

And settings:

```json
   "go.lintTool": "gometalinter"
```

Though it matches with the output of every CLI reporting the issue it seems incorrect from the standpoint of the user, I should only be told about this error once. I would expect multiple errors for the same line at the same offset to get rolled up into one, especially if the error message is identical.

_Copied from original issue: Microsoft/vscode-go#1013_

**Comment by ramya-rao-a** [comment]
_From @daenney on May 29, 2017 20:54_

Interestingly the spurious output in this specific case disappears if one sets `go.lintFlags` to `--fast` but it might be that it just happens to disable every other linter that tripped over this.

**Comment by ramya-rao-a** [comment]
@daenney I'd prefer to get a fix for this from `gometalinter` itself.
Maybe a flag `--removeDuplicates` that would return only unique errors.

@alecthomas thoughts?

**Comment by casimir** [comment]
Looks like you're looking for the `--aggregate` flag introduced in #180.

**Comment by daenney** [comment]
Fabulous, I had completely missed that.

@ramya-rao-a Can we use that as a default argument for gometalinter in vscode-go?

**Comment by ramya-rao-a** [comment]
Thanks @casimir !

@daenney Sure we can, I'll re-open the original issue
